subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -0,0 +1,94 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrading_version_8]]
+= Upgrading your build from Gradle 8.x to the latest
+
+This chapter provides the information you need to migrate your Gradle 8.x builds to the latest Gradle release. For migrating from Gradle 4.x, 5.x, 6.x, or 7.x, see the <<upgrading_version_7.adoc#upgrading_version_7, older migration guide>> first.
+
+We recommend the following steps for all users:
+
+. Try running `gradle help --scan` and view the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan.
++
+image::deprecations.png[Deprecations View of a Gradle Build Scan]
++
+This is so that you can see any deprecation warnings that apply to your build.
++
+Alternatively, you could run `gradle help --warning-mode=all` to see the deprecations in the console, though it may not report as much detailed information.
+. Update your plugins.
++
+Some plugins will break with this new version of Gradle, for example because they use internal APIs that have been removed or changed. The previous step will help you identify potential problems by issuing deprecation warnings when a plugin does try to use a deprecated part of the API.
++
+. Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
+. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.
+
+[[changes_9.0]]
+== Upgrading from 8.x and earlier
+
+=== Warnings that are now errors
+
+[[deprecated_configurations_should_not_be_used]]
+==== Usage of deprecated configurations is now an error
+
+Usage of the following `@link:{javadocPath}/javadoc/org/gradle/api/internal/artifacts/configurations/ConfigurationRoles.htm[ConfigurationRoles]` are now errors:
+
+- `LEGACY`
+- `DEPRECATED_CONSUMABLE`
+- `DEPRECATED_RESOLVABLE`
+
+=== Removed APIs
+
+=== Potential breaking changes
+
+=== Deprecations
+
+[[custom_configuration_roles]]
+Custom roles have been deprecated. Use pre-defined roles from `@link:{javadocPath}/javadoc/org/gradle/api/internal/artifacts/configurations/ConfigurationRoles.htm[ConfigurationRoles]` instead.
+
+[[reserved_configuration_names]]
+The `java-base` plugin will create configurations with for every sourceset using these patterns:
+
+ - `<SourceSet Name>Implementation`
+ - `<SourceSet Name>CompileOnly`
+ - `<SourceSet Name>RuntimeOnly`
+ - `<SourceSet Name>CompileClasspath`
+ - `<SourceSet Name>RuntimeClasspath`
+ - `<SourceSet Name>AnnotationProcessor`
+ - `<SourceSet Name>RuntimeElements`
+
+In addition, the `java-library` plugin will create configurations with these patterns:
```

Comment:
The java library plugin does not create configuration for source sets other than the main source set. It only creates the `api` and `compileOnlyApi` configurations. 

However, feature variants may create configurations in the format you've listed: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmVariantBuilder.java#L170-L176

Reply:
| Do we plan to update this now that we're closer to merging?

It's no longer needed as a link, so I'll just remove it.